### PR TITLE
Update storage.md with Oracle Cloud Infrastructure provider

### PR DESCRIPTION
**Pull request by aarontams**
* [ ] I added CHANGELOG entry for this change.
* [ ] Change is not relevant to the end user.

## Changes
Issue: https://github.com/thanos-io/thanos/issues/5592

This PR sync up https://github.com/thanos-io/thanos/blob/main/docs/storage.md and https://github.com/thanos-io/objstore/blob/main/README.md. See issue for details.

## Verification

This is just documentation sync up. No test is needed.

### CODE REVIEW

**Review by yeya24 — APPROVED** [review]
Thanks! Can you sign up the DCO?

**Comment by aarontams** [comment]
> Thanks! Can you sign up the DCO?

Sorry, forgot about it.
Done.

File: docs/storage.md [hunk 1]
```diff
@@ -493,6 +494,72 @@ config:
 prefix: ""
 ```
 
+### Oracle Cloud Infrastructure Object Storage
+
+To configure Oracle Cloud Infrastructure (OCI) Object Storage as Thanos Object Store, you need to provide appropriate authentication credentials to your OCI tenancy. The OCI object storage client implementation for Thanos supports either the default keypair or instance principal authentication.
+
+#### API Signing Key
+
+The default API signing key authentication provider leverages same [configuration as the OCI CLI](https://docs.oracle.com/en-us/iaas/Content/API/Concepts/cliconcepts.htm) which is usually stored in at `$HOME/.oci/config` or via variable names starting with the string `OCI_CLI`. You can also use environment variables that start with `TF_VAR`. If the same configuration is found in multiple places the provider will prefer the first one.
```
Comment (squat) [inline docs/storage.md:503]: > You can also use environment variables that start with `TF_VAR`.

This sounds funny to me. What does Thanos's OCI storage implementation have to do with Teratogen
Terraform?

Comment (aarontams) [reply docs/storage.md:503]: @squat 
Where did you see Teratogen? Can you clarify? Thx!

Comment (matej-g) [reply docs/storage.md:503]: I'll assume @squat had an autocorrect mishap, and he meant simply Terraform :), I assume he refers to why we're mentioning the `TF_VAR` here.

Comment (aarontams) [reply docs/storage.md]: Thanks @squat and @matej-g for the eagle eyes.
Updated my PR. Please review again.

**Review by matej-g — APPROVED** [review]
This looks good to me now, thanks @aarontams!

**Pull request merged**